upup/pkg/fi/cloudup/azure/azure_cloud.go

```diff
@@ -136,39 +139,82 @@ func (c *azureCloudImplementation) GetApiIngressStatus(cluster *kops.Cluster) ([
 	var ingresses []kops.ApiIngressStatus
 	var rg string = cluster.AzureResourceGroupName()
 
-	// Get scale sets in cluster resource group and find masters scale set
-	scaleSets, err := c.vmscaleSetsClient.List(context.TODO(), rg)
-	if err != nil {
-		return nil, fmt.Errorf("error getting Cluster Master Scale Set for API Ingress Status: %s", err)
-	}
-	var vmssName string
-	for _, scaleSet := range scaleSets {
-		val, ok := scaleSet.Tags[TagClusterName]
-		val2, ok2 := scaleSet.Tags[TagNameRolePrefix+TagRoleMaster]
-		if ok && *val == cluster.Name && ok2 && *val2 == "1" {
-			vmssName = *scaleSet.Name
-			break
+	lbSpec := cluster.Spec.API.LoadBalancer
+	if lbSpec != nil {
+		// Get loadbalancers in cluster resource group
+		lbs, err := c.loadBalancersClient.List(context.TODO(), rg)
+		if err != nil {
+			return nil, fmt.Errorf("error getting Loadbalancer for API Ingress Status: %s", err)
 		}
-	}
-	if vmssName == "" {
-		return nil, fmt.Errorf("error getting Master Scale Set Name for API Ingress Status")
-	}
 
-	// Get masters scale set network interfaces and append to api ingress status
-	nis, err := c.NetworkInterface().ListScaleSetsNetworkInterfaces(context.TODO(), rg, vmssName)
-	if err != nil {
-		return nil, fmt.Errorf("error getting Master Scale Set Network Interfaces for API Ingress Status: %s", err)
-	}
-	for _, ni := range nis {
-		if !*ni.Primary {
-			continue
+		for _, lb := range lbs {
+			val := lb.Tags[TagClusterName]
+			if val == nil || *val != cluster.Name {
+				continue
+			}
+			for _, i := range *lb.LoadBalancerPropertiesFormat.FrontendIPConfigurations {
+				if lb.LoadBalancerPropertiesFormat == nil || lb.LoadBalancerPropertiesFormat.FrontendIPConfigurations == nil {
+					continue
+				}
```

Comment:
We should ensure `LoadBalancerPropertiesFormat` is not nil before the range loop. `FrontendIPConfigurations` is okay to be nil, the range loop will treat it as an empty collection.
 
```suggestion
			if lb.LoadBalancerPropertiesFormat == nil {
				continue
			}
			for _, i := range lb.LoadBalancerPropertiesFormat.FrontendIPConfigurations {
```

Reply:
made the change.